Add KorSTS and KLUE-STS datasets

taeminlee opened the pull request:
## Checklist for adding MMTEB dataset

Reason for dataset addition:

KorSTS and KLUE-STS are Korean STS test sets. Most Korean embedding STS research uses these datasets for benchmarking.

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
      - [x] KorSTS
        `{'cos_sim': {'pearson': 0.7383478084364756, 'spearman': 0.744222458287421}, 'manhattan': {'pearson': 0.7250048454996945, 'spearman': 0.7191512876121664}, 'euclidean': {'pearson': 0.7239498937992224, 'spearman': 0.7181598809265461}, 'evaluation_time': 1.64}`
    - [x] KLUE-STS
      - `{'cos_sim': {'pearson': 0.6643426782389475, 'spearman': 0.6589523623849758}, 'manhattan': {'pearson': 0.6482246986840657, 'spearman': 0.6441548929177391}, 'euclidean': {'pearson': 0.6462564265558833, 'spearman': 0.6417309818469817}, 'evaluation_time': 1.16}`
  - [x] `intfloat/multilingual-e5-small`
      - [x]  KorSTS
          - `{'cos_sim': {'pearson': 0.7523126858440565, 'spearman': 0.7451242052419457}, 'manhattan': {'pearson': 0.7444275035327651, 'spearman': 0.7443553691832091}, 'euclidean': {'pearson': 0.74503523721575, 'spearman': 0.7451274994032616}, 'evaluation_time': 1.61}`
      - [x] KLUE-STS
        - `{'cos_sim': {'pearson': 0.7358337734443828, 'spearman': 0.7432746758420844}, 'manhattan': {'pearson': 0.7444151021204608, 'spearman': 0.7433643088238561}, 'euclidean': {'pearson': 0.7441718455345613, 'spearman': 0.7432746758420844}, 'evaluation_time': 1.14}`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [POINTS.md](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) file.

isaac-chung commented on mteb/tasks/STS/kor/KlueSTS.py:
```diff
@@ -0,0 +1,56 @@
+        eval_splits=["validation"],
+        eval_langs=["kor-Hang"],
+        main_score="cosine_spearman",
+        date=None,
```
Could we please add a rough estimate? e.g. ("2021-01-01", "{paper date}")

taeminlee replied:
I've added an estimated dates based on the KLUE and Kor-STS papers. :)

isaac-chung commented on mteb/tasks/STS/kor/KlueSTS.py:
```diff
@@ -0,0 +1,56 @@
+        eval_langs=["kor-Hang"],
+        main_score="cosine_spearman",
+        date=None,
+        form=None,
+        domains=None,
+        task_subtypes=None,
```
Please fill these in according to https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md

taeminlee replied:
I was looking for information on how to enter metadata, thanks for the heads up. I added the form and domain by referencing the paper. I left the subtype as None because it doesn't apply.

isaac-chung commented on mteb/tasks/STS/kor/KlueSTS.py:
```diff
@@ -0,0 +1,56 @@
+        domains=None,
+        task_subtypes=None,
+        license="CC-BY-SA-4.0",
+        socioeconomic_status=None,
+        annotations_creators=None,
```
Same as above please.

taeminlee replied:
found that there is an annotation step only for KLUE-STS. added that information.

isaac-chung commented on mteb/tasks/STS/kor/KlueSTS.py:
```diff
@@ -0,0 +1,56 @@
+        license="CC-BY-SA-4.0",
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=None,
```
`[]` means no dialects.
```suggestion
        dialect=[],
```

taeminlee replied:
changed

isaac-chung commented on mteb/tasks/STS/kor/KlueSTS.py:
```diff
@@ -0,0 +1,56 @@
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=None,
+        text_creation=None,
```
Same as above please.

taeminlee replied:
added information for text_creation.

isaac-chung reviewed (changes requested):
Thanks for the contribution! Added a few comments below.
In addition, 
1. Instead of including results in the PR description, could you please run `mteb run -m {model_name} -t {task_name}` and commit those results? They would be generated as json files in the right folder.
2. Could you please update the points as 2+4 (first Korean STS dataset) + 2 = 8 for yourself, and +1 for myself (`isaac-chung`) for the review?

KennethEnevoldsen commented:
@taeminlee, please also submit the models scores along with the PR

KennethEnevoldsen commented on mteb/tasks/STS/kor/KlueSTS.py:
```diff
@@ -0,0 +1,56 @@
+      primaryClass={cs.CL}
+}""",
+        n_samples={"validation": 519},
+        avg_character_length=None,
```
Should be added as well

KennethEnevoldsen commented on mteb/tasks/STS/kor/KorSTS.py:
```diff
@@ -0,0 +1,46 @@
+  year={2020}
+}""",
+        n_samples={"test": 1379},
+        avg_character_length=None,
```
same here

taeminlee replied:
added

KennethEnevoldsen commented on mteb/tasks/STS/kor/KorSTS.py:
```diff
@@ -0,0 +1,46 @@
+            "path": "dkoterwa/kor-sts",
+            "revision": "016f35f9b961daaaa7a352e927084e3da662ac1f",
+        },
+        description="",
```
Please add the description as well.

KennethEnevoldsen commented on mteb/tasks/STS/kor/KlueSTS.py:
```diff
@@ -0,0 +1,56 @@
+            "name": "sts",
+            "revision": "349481ec73fff722f88e0453ca05c77a447d967c",
+        },
+        description="",
```
Please add the description here as well

taeminlee replied:
added

taeminlee commented:
> Thanks for the contribution! Added a few comments below. In addition,
> 
> 1. Instead of including results in the PR description, could you please run `mteb run -m {model_name} -t {task_name}` and commit those results? They would be generated as json files in the right folder.

I added the run results as you suggested. Thanks for letting know.

> 2. Could you please update the points as 2+4 (first Korean STS dataset) + 2 = 8 for yourself, and +1 for myself (`isaac-chung`) for the review?

Oh thank you for finding the hidden score. I've also added a point for isaac-chung.

taeminlee commented:
> @taeminlee, please also submit the models scores along with the PR

Yes, we've included the model's scores. Also I've fixed and committed both reviews from @isaac-chung and @KennethEnevoldsen.
